source/includes/monitoring/sdam.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+require __DIR__ . '/vendor/autoload.php';
+
+// start-mysubscriber
+class MySubscriber implements MongoDB\Driver\Monitoring\SDAMSubscriber
+{
+    private $stream;
+    public function __construct($stream)
+    {
+        $this->stream = $stream;
+    }
+    public function serverOpening(MongoDB\Driver\Monitoring\ServerOpeningEvent $event): void
+    {
+        fwrite($this->stream, sprintf(
+            'Server opening on %s:%s%s',
+            $event->getHost(),
+            $event->getPort(),
+            PHP_EOL,
+        ));
```

Comment:
```suggestion
        fprintf(
            $this->stream,
            "Server opening on %s:%s\n",
            $event->getHost(),
            $event->getPort(),
        );
```

See: https://www.php.net/fprintf

Reply:
Good to know! I grabbed that code from here intially: https://www.mongodb.com/docs/php-library/current/reference/method/MongoDBClient-addSubscriber/#example

Reply:
Looks like @GromNaN originally wrote that in https://github.com/mongodb/mongo-php-library/commit/b455ab7c20ce337653efe39afed19b8a37c0ff5a :smile: